### PR TITLE
feature/imc mockup

### LINKED ISSUE #140: Create IMC mockup
Create a mockup for the backend that should be used throught the rest of the development time until the IMC message system is implemented from SINTEF in the backend.

Features:

- Create a pop-up window where you can choose mode and the values for each mode
- Send the data over TCP
- Encode and decode the data to mockup the ROV
- Show the input IMC message
- Change the message you're sending

### PR DESCRIPTION
**Pull request by greenbech**
This PR implements a tcp server, that allows us to mock up the ROV for development. It also implements the IMC protocol for manual mode (we should also implement this for the other modes).

To enable the mock up you have to do a couple of steps:
1. Go go `Simulator -> Mockup window -> Click "Start Server"`
_This starts the ROV server and it now waits for a connection. It should now display "Server is running..."_
2.  Go to `Rov -> Settings -> Change protocol to IMC (not OLD)`
3. Click `Rov -> "Connect to TCP"`

This is the GUI for the mock up server. It is possible the changes values we want to send to our GUI her
Skjermbilde 2019-10-24 22 54 41

Added a "Message Protocol" option in the settings menu.
Skjermbilde 2019-10-24 22 55 22

Kudos to @eirikdahlen for the front end part of this PR 🎉

This PR closes #140.

### CODE REVIEW

File: public/TCP/TCPClient.js [hunk 1]
```diff
@@ -10,34 +17,46 @@ const limitAttempts = 3;
 function getConnectedClient() {
   //console.log('Attempting to create TCP client and connect to server..');
   const client = new net.Socket();
+  messageProtocol = global.settings.messageProtocol;
+  console.log(messageProtocol);
```
Comment (Trebua) [inline public/TCP/TCPClient.js]: Kommenter bort log

File: public/TCP/TCPClient.js [hunk 2]
```diff
@@ -10,34 +17,46 @@ const limitAttempts = 3;
...
-      sway: 0.0,
-      heave: 0.0,
-      roll: 0.0,
-      pitch: 0.0,
-      yaw: 0.0,
-      autodepth: false,
-      autoheading: false,
-    });
+    console.log(`Client: connection established with server!`);
+    console.log(messageProtocol);
```
Comment (Trebua) [inline public/TCP/TCPClient.js]: Kommenter ut logs, det fucker med testene og brukes ikke i prod

File: public/utils/windows.js [hunk 3]
```diff
@@ -138,6 +138,34 @@ function createXboxMappingWindow() {
   xboxMappingWindow.setMenu(null);
 }
 
+function createMockupWindow() {
+  let mockupWindow = new electron.BrowserWindow({
```
Comment (Trebua) [inline public/utils/windows.js]: remove electron.

File: src/MockupComponents/ModeAvailableToggles.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import './css/ModeAvailableToggles.css';
+
+const { ipcRenderer } = require('electron');
+
+export default function ModeAvailableToggles() {
+  const [nfAvailable, setNfAvailable] = useState(true);
+  const [dpAvailable, setDpAvailable] = useState(true);
+
+  function toggleDPAvailable() {
+    // ipcRenderer.send('rov-mock-up-send-df-available', !dpAvailable);
```
Comment (Trebua) [inline src/MockupComponents/ModeAvailableToggles.jsx]: remove if not in use

File: src/MockupComponents/ModeAvailableToggles.jsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    setDpAvailable(!dpAvailable);
+  }
+  useEffect(() => {
+    ipcRenderer.send('rov-mock-up-send-df-available', dpAvailable);
+  }, [dpAvailable]);
+
+  function toggleNFAvailable() {
+    // ipcRenderer.send('rov-mock-up-send-nf-available', !nfAvailable);
```
Comment (Trebua) [inline src/MockupComponents/ModeAvailableToggles.jsx]: remove if not in use

File: src/MockupComponents/ROVMockUp.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+export default function ROVMockUp() {
+  const [mode, setMode] = useState(modeEnum.MANUAL);
+  const [recievedData, setRecievedData] = useState(null);
+  const [isServerRunning, setIsServerRunning] = useState(false);
+
+  // TODO: add this functionality in TCPServerMockUp
+  function startServer() {
+    if (!isServerRunning) {
+      console.log('hei');
```
Comment (Trebua) [inline src/MockupComponents/ROVMockUp.jsx]: remove 😺

File: src/MockupComponents/css/NFView.css [hunk 7]
```diff
@@ -0,0 +1,32 @@
+.NFViewRoot {
+  display: flex;
+  flex-direction: row;
+}
+
+.NFViewRoot > div {
+  background-color: lightsteelblue;
```
Comment (Trebua) [inline src/MockupComponents/css/NFView.css]: burde kanskje bruke samme farger som kontrollvinduet her? altså farger definert i index.css

File: public/TCP/TCPClient.js [hunk 8]
```diff
@@ -79,4 +98,66 @@ function sendData(client, data) {
...
+  */
+  let desiredControlBuf = encode.desiredControl({
+    x: global.toROV.surge,
+    y: global.toROV.sway,
+    z: global.toROV.heave,
+    k: 0.0,
+    m: global.toROV.pitch,
+    n: global.toROV.yaw,
+    flags: {
+      x: true,
```
Comment (hsyver) [inline public/TCP/TCPClient.js]: Er ikke dette omvendt av hva vi har definert i imc-forslaget?

File: public/TCP/TCPClient.js (hunk 1 above)
Comment (eirikdahlen) [reply public/TCP/TCPClient.js]: Fixed, remember English!

File: public/TCP/TCPClient.js (hunk 2 above)
Comment (eirikdahlen) [reply public/TCP/TCPClient.js]: Line 29 could be nice to have, as it is visible in the terminal when starting the server.
Removed line 30 👌🏼

File: src/MockupComponents/css/NFView.css (hunk 7 above)
Comment (eirikdahlen) [reply src/MockupComponents/css/NFView.css]: Fixed!

**Pull request merged**